File: Easy/2704-maximum-difference-by-remapping-a-digit/2704-maximum-difference-by-remapping-a-digit.cpp
class Solution {
public:
    int toNum(const vector<int>& arr) {
        int n = 0;
        for (int it : arr)
            n = n * 10 + it;
        return n;
    }

    int minMaxDifference(int num) {
        vector<int> digits;
        int temp = num;

        string s = to_string(temp);
        for (char c : s) {
            digits.push_back(c - '0');
        }

        vector<int> maxDigits = digits;
        for (int i = 0; i < digits.size(); i++) {
            if (digits[i] != 9) {
                int x = digits[i];
                for (int& d : maxDigits) {
                    if (d == x) 
                        d = 9;
                }
                break;
            }
        }

        vector<int> minDigits = digits;
        for (int i = 0; i < digits.size(); i++) {
            if (digits[i] != 0) {
                int x = digits[i];
                for (int& d : minDigits) {
                    if (d == x) 
                        d = 0;
                }
                break;
            }
        }

        int maxi = toNum(maxDigits), mini = toNum(minDigits);

        return maxi - mini;
    }
};

File: Easy/2704-maximum-difference-by-remapping-a-digit/2704-maximum-difference-by-remapping-a-digit.py
class Solution:
    def minMaxDifference(self, num: int) -> int:
        digits = list(map(int, str(num)))

        for d in digits:
            if d != 9:
                max_digits = [9 if x == d else x for x in digits]
                break
        else:
            max_digits = digits[:] 

        for d in digits:
            if d != 0:
                min_digits = [0 if x == d else x for x in digits]
                break
        else:
            min_digits = digits[:]

        max_num = int(''.join(map(str, max_digits)))
        min_num = int(''.join(map(str, min_digits)))

        return max_num - min_num